src/datasets/arrow_dataset.py

```diff
@@ -4666,6 +4666,42 @@ def to_dict(self, batch_size: Optional[int] = None, batched: bool = False) -> Un
                 for offset in range(0, len(self), batch_size)
             )
 
+    def to_list(self, batch_size: Optional[int] = None, batched: bool = False) -> Union[list, Iterator[list]]:
```

Comment:
The order of the arguments in the function signature don't match the order in the docstring
```suggestion
    def to_list(self, batched: bool = False, batch_size: Optional[int] = None) -> Union[list, Iterator[list]]:
```

Reply:
This is for consistency with `Dataset.to_dict`